### ISSUE TITLE
Crash with misaligned address in 2d simulations

### ISSUE THREAD

**Issue by sbastrakov** [ComputationalRadiationPhysics/picongpu#3891]
It seems we recently introduced some bug which results in alpaka/cupla exception ` 'cudaErrorMisalignedAddress': 'misaligned address'!` (or something disguised as it). Attaching a 2d setup which for me consistently reproduces the bug on fwk394. The 3d version of it works, but i do not know if by accident or not. Changing grid size and number of particles per cell doesn't help.
[particleInitializationBug.zip](https://github.com/ComputationalRadiationPhysics/picongpu/files/7426327/particleInitializationBug.zip)

The crash happens on `dev` branch, in [kernel creating particles](https://github.com/ComputationalRadiationPhysics/picongpu/blob/f73109281b234b1d0c8eb646e44d680d5f7ce95d/include/picongpu/particles/Particles.tpp#L418) so when the particle data structure is populated.
Here is full error output with max verbosity:
```
(verbose output as usual)
PIConGPUVerbose SIMULATION_STATE(16) | initialize density profile for species e
[cupla] Error: </home/bastra54/src/picongpu/include/pmacc/../pmacc/eventSystem/events/kernelEvents.hpp>:165 Last error after kernel launch N8picongpu27KernelFillGridWithParticlesILj64ENS_9ParticlesIN5pmacc4meta6StringIJLc101EEEEN5boost3mpl6v_itemINS_11chargeRatioINS_20ChargeRatioElectronsENS2_13pmacc_isAliasEEENS8_INS_9massRatioINS_18MassRatioElectronsESB_EENS8_INS_7currentINS_13currentSolver9EsirkepovINS_9particles6shapes3TSCENSH_8strategy16CachedSupercellsELj2EEESB_EENS8_INS_13interpolationINS_28FieldToParticleInterpolationISL_NS_30AssignedTrilinearInterpolationEEESB_EENS8_INS_5shapeISL_SB_EENS8_INS_14particlePusherINSJ_6pusher5BorisESB_EENS7_7vector0IN4mpl_2naEEELi0EEELi0EEELi0EEELi0EEELi0EEELi0EEENS8_INS_9weightingENS8_INS_8momentumENS8_INS_8positionINS_12position_picESB_EES14_Li0EEELi0EEELi0EEEEEEE [/home/bastra54/src/picongpu/include/pmacc/../picongpu/particles/Particles.tpp:418 ]
[cupla] Error: </home/bastra54/src/picongpu/include/pmacc/../pmacc/eventSystem/events/kernelEvents.hpp>:165 Last error after kernel launch N8picongpu27KernelFillGridWithParticlesILj64ENS_9ParticlesIN5pmacc4meta6StringIJLc101EEEEN5boost3mpl6v_itemINS_11chargeRatioINS_20ChargeRatioElectronsENS2_13pmacc_isAliasEEENS8_INS_9massRatioINS_18MassRatioElectronsESB_EENS8_INS_7currentINS_13currentSolver9EsirkepovINS_9particles6shapes3TSCENSH_8strategy16CachedSupercellsELj2EEESB_EENS8_INS_13interpolationINS_28FieldToParticleInterpolationISL_NS_30AssignedTrilinearInterpolationEEESB_EENS8_INS_5shapeISL_SB_EENS8_INS_14particlePusherINSJ_6pusher5BorisESB_EENS7_7vector0IN4mpl_2naEEELi0EEELi0EEELi0EEELi0EEELi0EEELi0EEENS8_INS_9weightingENS8_INS_8momentumENS8_INS_8positionINS_12position_picESB_EES14_Li0EEELi0EEELi0EEEEEEE [/home/bastra54/src/picongpu/include/pmacc/../picongpu/particles/Particles.tpp:418 ]
Unhandled exception of type 'St13runtime_error' with message '[cupla] Error: not defined cuplaError', terminating
terminate called after throwing an instance of 'std::runtime_error'
  what():  /home/bastra54/src/picongpu/thirdParty/cupla/alpaka/include/alpaka/event/EventUniformCudaHipRt.hpp(201) 'cudaEventSynchronize(event.m_spEventImpl->m_UniformCudaHipEvent)' returned error  : 'cudaErrorMisalignedAddress': 'misaligned address'!
```

Perhaps related to #3861 or #3841 but hard to tell. Perhaps last changes to mallocMC broke it, but that's only guessing as it is related to memory allocation. I will try on a PIConGPU version before the mallocMC update.

**Comment by sbastrakov** [comment]
Issue in mallocMC is fixed, now we need to update the version used by PIConGPU.

**Comment by sbastrakov** [comment]
Solved with #3897.
